docs/blueprints/referral-manager-v1.md

```diff
@@ -0,0 +1,206 @@
+
+We want referrals to be valid on all Tardigrade satellites. This new design will allow the user to select any satellite, regardless of the satellite the referral was generated for.
+
+### Non-Goals
+A registration page where a user can select their preferred Satellite during registration.
+
+## Design
+
+### Definitions
+
+Referral Manager
+: A standalone process running separately from satellites. Only Tardigrade-branded satellites should be able to talk to it.
+
+Referral Manager CLI
+: A CLI that will allow operators to invoke referral link generation.
+
+Invitation Token
+: A random number generated by Referral Manager for each user in the referral program.
+
+Referral Link
+: A one-time-use URL that contains a unique invitation token.
+
+Eligible Users
+: Users who have fewer than the desired number of unredeemed referral links.
+
+Owner's Satellite
+: The satellite that the owner of a particular token belongs to.
+
+Redeemed Satellite
+: The satellite where the new user redeeming a token has registered on.
+
+### How it will work
+
+_User Interface_
+
+1. Users will have a referral tab on the UI. When clicked, it will trigger the satellite to send a request to retrieve referral links from the Referral Manager:
+    - The Referral Manager will attempt to generate and/or fetch unredeemed tokens in the `tokens` table for this user ID and satellite.
+        - If the user ID doesn't exist in the Referral Manager's `users` table, the Referral Manager will add an entry for this user ID and satellite in the `users` table and return an empty response to the satellite.
+        - If the `users` table contains a value larger than `0` for `new_tokens` for this user ID, the Referral Manager will generate that number of new tokens, add the new unredeemed tokens to the `tokens` table, and set the value of `new_tokens` for that user to `0` in the `users` table.
+        - If there are unredeemed tokens in the `tokens` table, the Referral Manager will respond to the satellite's request with the tokens after generating new tokens. Otherwise, it will return an empty response.
+    - The satellite receives the response from Referral Manager.
+        - If the response payload contains tokens, the satellite will display them in the UI.
+        - If it's an empty response payload, the satellite will display a message `No available referral links. Try again later.`
+
+
+_Referral Manager CLI_
+
+1. `referral-manager start --tokens-per-user=3 <satelliteURL1> <satelliteURL2> ...` initiates the invitation token generation process. The `--dry-run` flag can optionally be added to run the process without actually generating tokens. The dry run allows for estimating how many new tokens would be created by the command.
+2. The Referral Manager queries both `users` and `tokens` tables for all users where `new_tokens + unredeemed_tokens < flags.tokens_per_user` and where the satellite matches one of the CLI-provided satellite URLs. These are the "eligible users".
```

Comment:
You mentioned that we want to add another flag limiting the maximum number of unredeemed tokens per user, which would also come into play here.